src/plugins/timeseries-plugin.ts

```diff
@@ -0,0 +1,270 @@
+        super.init(renderEngine, interactionsEngine);
+
+        this.interactionsEngine.on('change-position', this.handlePositionChange.bind(this));
+        this.interactionsEngine.on('hover', this.handleHover.bind(this));
+
+        this.interactionsEngine.on('up', this.handleMouseUp.bind(this));
+    }
+
+    handlePositionChange(position: { deltaX: number }) {
+        const startPositionX = this.renderEngine.parent.positionX;
+
+        this.interactionsEngine.setCursor('grabbing');
+
+        this.renderEngine.tryToChangePosition(position.deltaX);
+
+        if (startPositionX !== this.renderEngine.parent.positionX) {
+            this.renderEngine.parent.render();
+        }
+    }
+
+    handleMouseUp() {
+        this.interactionsEngine.clearCursor();
+    }
+
+    setData(data: TimeseriesPoint[]) {
+        this.data = data;
+
+        let min = Number.MAX_VALUE;
+        let max = Number.MIN_VALUE;
+        let first = Number.MAX_VALUE;
+        let last = Number.MIN_VALUE;
+
+        this.data.forEach(([timestamp, value]) => {
+            if (value < min) {
+                min = value;
+            }
+            if (value > max) {
+                max = value;
+            }
+
+            if (timestamp < first) {
+                first = timestamp;
+            }
+            if (timestamp > last) {
+                last = timestamp;
+            }
+        });
+
+        this.summary = {
+            min,
+            max,
+            first,
+            last,
+        };
+    }
+
+    calcRect(start: number, duration: number, isEnd: boolean) {
+        const w = duration * this.renderEngine.zoom;
+
+        return {
+            x: this.renderEngine.timeToPosition(start),
+            w: isEnd ? (w <= 0.1 ? 0.1 : w >= 3 ? w - 1 : w - w / 3) : w,
+        };
+    }
+
+    handleHover(region: HitRegion<number> | null) {
+        this.hoveredRegion = region;
+    }
+
+    override renderTooltip() {
+        if (this.hoveredRegion) {
+            const round = (value: number) => Math.round(value * 100) / 100;
+            const data = this.hoveredRegion.data as TimeseriesPointWithExtra;
+
+            this.renderEngine.renderTooltipFromData(
+                [
+                    { text: `Value: ${round(data.value)}` },
+                    {
+                        text: `Timestamp: ${round(data.timestamp)}ms`,
+                    },
+                ],
+                this.interactionsEngine.getGlobalMouse()
+            );
+            return true;
+        }
+        return false;
+    }
+
+    private normalizeValue(value: number, heightPerValueUnit: number) {
+        return this.height - value * heightPerValueUnit;
+    }
+
+    private convertDataPointToPointWithIndex(
+        idx: number,
+        timestamp: number,
+        value: number,
+        heightPerValueUnit: number
+    ): TimeseriesPointWithExtra {
+        return {
```

Comment:
Why are you creating this object? You're just wrapping values (idx, timestamp, value) in an object, which makes no sense.(